src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  let obj = Object.assign({}, state);
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        obj = { ...obj, ...extraData };
+        res.push({ ...obj });
+        break;
+
+      case 'removeProperties':
+        for (const removeElement of keysToRemove) {
+          delete obj[removeElement];
+        }
+        res.push({ ...obj });
+        break;
+
+      case 'clear':
+        obj = {};
+        res.push({ ...obj });
```

Comment:
yes, but you are doing it in every cases so move it after all cases by DRY principle